uamqp/client.py

```diff
@@ -647,13 +647,13 @@ def _client_run(self):
         """
         # pylint: disable=protected-access
         self.message_handler.work()
+        self._connection.work()
```

Comment:
I think this change is probably fine. The only thing I can think that we might need to look out for is the error pattern. Are there any exceptions that can be raised in the `work` function, that previously would have been called after the pending messages had been processed, and will now be raised before (therefore not allowing pending messages to be processed)?

Reply:
nope, the processing of pending messages doesn't raise error, the exception would be caught and set on the message status.

https://github.com/Azure/azure-uamqp-python/blob/7ee656a6959835d1b1267fb0f726e38567f4ed0e/uamqp/client.py#L633-L636

so basically the change is transparent to upper application, and then connection error shall be bubbled.
actually it's an improvement on the work flow that when the connection socket fails, the sender should not send at all and that's what we're already doing on the receiving side:

https://github.com/Azure/azure-uamqp-python/blob/7ee656a6959835d1b1267fb0f726e38567f4ed0e/uamqp/client.py#L953-L961

Reply:
Sounds good to me :)